R/zchunk_LB171.LC_R_Cropland_Yh_GLU_irr.R

```diff
@@ -51,82 +51,48 @@ module_aglu_LB171.LC_R_Cropland_Yh_GLU_irr <- function(command, ...) {
+      left_join(IrrRfdCropland, by=idvars) %>%
       group_by(GCAM_region_ID, GCAM_commodity, GLU) %>%
-      mutate(value = approx_fun(year, value)) %>%
-      ungroup() -> rfdHarvCropLand_interp
-
-    # Calculate economic yields for irrigated crops, as production divided by cropland
-    L161.ag_irrProd_Mt_R_C_Y_GLU %>%
-      # Irrigated production in Mt by GCAM region, commodity, year, and GLU
-      rename(prod = value) %>%
-      # Match with the irrigated cropland cover in bm2 by GCAM region, commodity, year, and GLU
-      left_join_error_no_match(irrHarvCropLand_interp, by = c("GCAM_region_ID", "GCAM_commodity", "GLU", "year")) %>%
-      # Calculate economic yields as irrigated production divided by irrigated cropland, Mt/bm2 = kg/m2
-      mutate(value = prod / value,
-             # Replace missing value with zero
-             value = if_else(is.na(value), 0, value)) %>%
-      select(-prod) ->
-      L171.ag_irrEcYield_kgm2_R_C_Y_GLU
-
-    # Calculate economic yields for rainfed crops, as production divided by cropland
-    L161.ag_rfdProd_Mt_R_C_Y_GLU %>%
-      # Rainfed production in Mt by GCAM region, commodity, year, and GLU
-      rename(prod = value) %>%
-      # Match with the rainfed cropland cover in bm2 by GCAM region, commodity, year, and GLU
-      left_join_error_no_match(rfdHarvCropLand_interp, by = c("GCAM_region_ID", "GCAM_commodity", "GLU", "year")) %>%
-      # Calculate economic yields as rainfed production divided by rainfed cropland, Mt/bm2 = kg/m2
-      mutate(value = prod / value,
-             # Replace missing value with zero
-             value = if_else(is.na(value), 0, value)) %>%
-      select(-prod) ->
-      L171.ag_rfdEcYield_kgm2_R_C_Y_GLU
-
-    # Produce outputs
-    L171.LC_bm2_R_irrHarvCropLand_C_Yh_GLU %>%
+      mutate(irr.harvarea = approx_fun(year, irr.harvarea, rule=2),
+             rfd.harvarea = approx_fun(year, rfd.harvarea, rule=2)) %>%
+      ungroup -> IrrRfdCropland.interp
+
+    ## Compute economic yield for each category as production divided by
+    ## harvested area.
+    prod.both <-
+        full_join(rename(L161.ag_rfdProd_Mt_R_C_Y_GLU, prod.rfd = value),
+                  rename(L161.ag_irrProd_Mt_R_C_Y_GLU, prod.irr = value),
+                  by = idvars) %>%
+          filter(year %in% c(HISTORICAL_YEARS, FUTURE_YEARS))
+
+    ecyield.both <- left_join_error_no_match(prod.both, IrrRfdCropland.interp,
+                                             by = idvars) %>%
+      mutate(irr.yld = prod.irr / irr.harvarea,
+             irr.yld = replace(irr.yld, is.na(irr.yld), 0.0),
+             rfd.yld = prod.rfd / rfd.harvarea,
+             rfd.yld = replace(rfd.yld, is.na(rfd.yld), 0.0))
+
+
+    ## Produce outputs.  Note that for harvested area, we use the values without
+    ## interpolation, stored in IrrRfdCropland (as opposed to IrrRfdCropland.interp)
+    select(IrrRfdCropland, GCAM_region_ID, GCAM_commodity, GLU, year,
```

Comment:
How about `IrrRfdCropland %>% select(GCAM_region_ID, GCAM_commodity, GLU, year, value = irr.harvarea)` and below?